webviews/components/history_view.svelte

```diff
@@ -1,16 +1,175 @@
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED:
+        startTime = tsToDate(history.eventTime).getTime() / 1000
+        return ` ⇨ ${eventTypeName} (${history.workflowExecutionStartedEventAttributes?.workflowType?.name})`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
+        console.log(tsToDate(history.eventTime).getTime() / 1000, "-", startTime, "=", duration)
+        return ""
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_FAILED:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
+        // console.log((tsToDate(history.eventTime).getTime()/1000),"-",startTime,"=",duration)
+        return `⇨ ${eventTypeName}`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
+        // console.log((tsToDate(history.eventTime).getTime()/1000),"-",startTime,"=",duration)
+        return `⇨ ${eventTypeName}`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_CONTINUED_AS_NEW:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
+        // console.log((tsToDate(history.eventTime).getTime()/1000),"-",startTime,"=",duration)
+        return `⇨ ${eventTypeName}`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_SCHEDULED:
+        return ` ⇦ ${eventTypeName} (${history.activityTaskScheduledEventAttributes?.activityType?.name})`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_STARTED:
+        if (history.activityTaskStartedEventAttributes?.lastFailure) {
+          return ` ⇦ ${eventTypeName} (${history.activityTaskStartedEventAttributes?.lastFailure.message})`
+        } else {
+          return ` ⇦ ${eventTypeName}`
+        }
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED:
+        return ` ⇨ ${eventTypeName} (${history.workflowExecutionSignaledEventAttributes?.signalName})`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_TIMER_STARTED:
+        return `⇨ ⏱ ${eventTypeName} (${history.timerStartedEventAttributes?.startToFireTimeout?.seconds}s)`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_TIMER_FIRED:
+        return `⇦ ⏱ ${eventTypeName}`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_FAILED:
+        return ` ⇦ ${eventTypeName} (${history.activityTaskFailedEventAttributes?.failure})`
+      case temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED:
+        duration = +(tsToDate(history.eventTime).getTime() / 1000 - startTime).toFixed(3)
+        // console.log((tsToDate(history.eventTime).getTime()/1000),"-",startTime,"=",duration)
+        return `⇨ ${eventTypeName}`
+    }
+    return `⇦ ${eventTypeName}`
```

Comment:
We could do further customizations here for child workflows and such, please add a TODO comment and let's try to move on.